Patch by Chris Waters and Sergei Golovan.

It sets paths which are normally point to a directory with Tcl
sources to a subdirectory of /usr/include/tcl8.4 (debian/rules puts
private Tcl headers there).

Also, it allows to find tclConfig.sh in /usr/share/tcltk/tcl8.4 and tkConfig.sh
in /usr/share/tcltk/tk8.4 where they are located in Debian installation.

--- tcl8.4-8.4.19.orig/unix/tcl.m4
+++ tcl8.4-8.4.19/unix/tcl.m4
@@ -83,6 +83,7 @@
 			`ls -d ${prefix}/lib 2>/dev/null` \
 			`ls -d /usr/local/lib 2>/dev/null` \
 			`ls -d /usr/contrib/lib 2>/dev/null` \
+			`ls -d /usr/share/tcltk/tcl8.4 2>/dev/null` \
 			`ls -d /usr/lib 2>/dev/null` \
 			; do
 		    if test -f "$i/tclConfig.sh" ; then
@@ -203,6 +204,7 @@
 			`ls -d ${prefix}/lib 2>/dev/null` \
 			`ls -d /usr/local/lib 2>/dev/null` \
 			`ls -d /usr/contrib/lib 2>/dev/null` \
+			`ls -d /usr/share/tcltk/tk8.4 2>/dev/null` \
 			`ls -d /usr/lib 2>/dev/null` \
 			; do
 		    if test -f "$i/tkConfig.sh" ; then
--- tcl8.4-8.4.19.orig/unix/tclConfig.sh.in
+++ tcl8.4-8.4.19/unix/tclConfig.sh.in
@@ -111,7 +111,7 @@
 
 # String to pass to linker to pick up the Tcl library from its
 # build directory.
-TCL_BUILD_LIB_SPEC='@TCL_BUILD_LIB_SPEC@'
+TCL_BUILD_LIB_SPEC='@TCL_LIB_SPEC@'
 
 # String to pass to linker to pick up the Tcl library from its
 # installed directory.
@@ -146,7 +146,7 @@
 # different place than the directory containing the source files, this
 # points to the location of the sources, not the location where Tcl was
 # compiled.
-TCL_SRC_DIR='@TCL_SRC_DIR@'
+TCL_SRC_DIR='@includedir@/tcl-private'
 
 # List of standard directories in which to look for packages during
 # "package require" commands.  Contains the "prefix" directory plus also
@@ -164,14 +164,14 @@
 
 # String to pass to linker to pick up the Tcl stub library from its
 # build directory.
-TCL_BUILD_STUB_LIB_SPEC='@TCL_BUILD_STUB_LIB_SPEC@'
+TCL_BUILD_STUB_LIB_SPEC='@TCL_STUB_LIB_SPEC@'
 
 # String to pass to linker to pick up the Tcl stub library from its
 # installed directory.
 TCL_STUB_LIB_SPEC='@TCL_STUB_LIB_SPEC@'
 
 # Path to the Tcl stub library in the build directory.
-TCL_BUILD_STUB_LIB_PATH='@TCL_BUILD_STUB_LIB_PATH@'
+TCL_BUILD_STUB_LIB_PATH='@TCL_STUB_LIB_PATH@'
 
 # Path to the Tcl stub library in the install directory.
 TCL_STUB_LIB_PATH='@TCL_STUB_LIB_PATH@'
